packages/web3-eth-abi/src/coders/base/numbersLimits.ts

```diff
@@ -0,0 +1,505 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// this file precalculate all the numbers limits
+// Note: this could be simplified using ** operator inside a for loop, but babel does not handle it well
+// 	you can find more at: https://github.com/babel/babel/issues/13109 and https://github.com/web3/web3.js/issues/6187
+
+// Here is how the limits would be calculated with ** operator:
+// const numberLimits = new Map<string, { min: bigint; max: bigint }>();
+// for (let i = 8; i <= 256; i += 8) {
+// 	numberLimits.set(`uint${i}`, {
+// 		min: BigInt(0),
+// 		max: BigInt(2) ** BigInt(i) - BigInt(1),
+// 	});
+// 	numberLimits.set(`int${i}`, {
+// 		min: -(BigInt(2) ** BigInt(i - 1)),
+// 		max: BigInt(2) ** BigInt(i - 1) - BigInt(1),
+// 	});
+// }
+
+export const numberLimits: Map<string, { min: bigint; max: bigint }> = new Map([
+	[
+		'uint8',
+		{
+			min: BigInt('0'),
+			max: BigInt('255'),
+		},
+	],
+	[
+		'int8',
+		{
+			min: BigInt('-128'),
+			max: BigInt('127'),
```

Comment:
Instead of adding this large file , why not to use function that you added `bigintPower` in existing Map code .
```
const numberLimits = new Map<string, { min: bigint; max: bigint }>();

// precalculate all the limits
for (let i = 8; i <= 256; i += 8) {
	numberLimits.set(`uint${i}`, {
		min: BigInt(0),
		max: BigInt(2) ** BigInt(i) - BigInt(1),
	});
	numberLimits.set(`int${i}`, {
		min: -(BigInt(2) ** BigInt(i - 1)),
		max: BigInt(2) ** BigInt(i - 1) - BigInt(1),
	});
}
```
and see to save some lib size: as in this PR its 

Total | 585 KB | 592 KB | 6.96 KB | 1.19%

Reply:
I though about reducing the execution time. But, sorry, I did not notice the size. I will modify to use the added `bigintPower`.

Reply:
I changed it to use a loop with multiplication instead of power.